src/Routers/FilesRouter.js

```diff
@@ -213,8 +227,18 @@ export class FilesRouter {
         // if the ParseFile returned is type uri, download the file before saving it
         await addFileDataIfNeeded(fileObject.file);
         // update fileSize
-        const bufferData = Buffer.from(fileObject.file._data, 'base64');
-        fileObject.fileSize = Buffer.byteLength(bufferData);
+        let fileData;
+        // if the file is a blob, get the size from the blob
+        if (typeof Blob !== 'undefined' && fileObject.file._source?.file instanceof Blob) {
```

Comment:
```suggestion
        if (fileObject.file._source?.file instanceof Blob) {
```

See https://github.com/parse-community/parse-server/pull/9286#discussion_r1791863516